Purchases/Purchasing/ProductsManager.swift

```diff
@@ -17,116 +17,43 @@ import StoreKit
-        guard identifiers.count > 0 else {
-            completion([])
-            return
-        }
-        queue.async { [self] in
-            let productsAlreadyCached = self.cachedProductsByIdentifier.filter { key, _ in identifiers.contains(key) }
-            if productsAlreadyCached.count == identifiers.count {
-                let productsAlreadyCachedSet = Set(productsAlreadyCached.values)
-                Logger.debug(String(format: Strings.offering.products_already_cached, identifiers))
-                completion(productsAlreadyCachedSet)
-                return
-            }
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    private(set) lazy var productsFetcherSK2 = ProductsFetcherSK2()
 
-            if let existingHandlers = self.completionHandlers[identifiers] {
-                Logger.debug(String(format: Strings.offering.found_existing_product_request, identifiers))
-                self.completionHandlers[identifiers] = existingHandlers + [completion]
-                return
-            }
-
-            Logger.debug(
-                String(format: Strings.offering.no_cached_requests_and_products_starting_skproduct_request, identifiers)
-            )
-            let request = self.productsRequestFactory.request(productIdentifiers: identifiers)
-            request.delegate = self
-            self.completionHandlers[identifiers] = [completion]
-            self.productsByRequests[request] = identifiers
-            request.start()
-        }
-    }
-
-    func cacheProduct(_ product: SKProduct) {
-        queue.async {
-            self.cachedProductsByIdentifier[product.productIdentifier] = product
-        }
+    init(productsRequestFactory: ProductsRequestFactory = ProductsRequestFactory()) {
+        self.productsFetcherSK1 = ProductsFetcherSK1(productsRequestFactory: productsRequestFactory)
     }
 
-}
-
-extension ProductsManager: SKProductsRequestDelegate {
-
-    func productsRequest(_ request: SKProductsRequest, didReceive response: SKProductsResponse) {
-        queue.async { [self] in
-            Logger.rcSuccess(Strings.network.skproductsrequest_received_response)
-            guard let requestProducts = self.productsByRequests[request] else {
-                Logger.error("requested products not found for request: \(request)")
-                return
-            }
-            guard let completionBlocks = self.completionHandlers[requestProducts] else {
-                Logger.error("callback not found for failing request: \(request)")
-                return
-            }
-
-            self.completionHandlers.removeValue(forKey: requestProducts)
-            self.productsByRequests.removeValue(forKey: request)
-
-            self.cacheProducts(response.products)
-            for completion in completionBlocks {
-                completion(Set(response.products))
+    func productsFromOptimalStore(withIdentifiers identifiers: Set<String>,
```

Comment:
For a second, I thought this API was literally a different store, like AppStore vs Play store. I'd be totally fine with an api that was more descriptive here, like `productsFromOptimalStoreKitVersion` or something like that.

Reply:
yeah, that's much better! I couldn't think of a name 😂